influxql/functions.go

```diff
@@ -357,6 +357,70 @@ func (r *IntegerMovingAverageReducer) Emit() []FloatPoint {
 	}
 }
 
+// FloatCumulativeSumReducer cumulates the values from each point.
+type FloatCumulativeSumReducer struct {
+	curr    FloatPoint
+	emitted bool
+}
+
+// NewFloatCumulativeSumReducer creates a new FloatCumulativeSumReducer.
+func NewFloatCumulativeSumReducer() *FloatCumulativeSumReducer {
+	return &FloatCumulativeSumReducer{
+		curr: FloatPoint{Nil: true},
+	}
+}
+
+func (r *FloatCumulativeSumReducer) AggregateFloat(p *FloatPoint) {
+	if !r.curr.Nil && r.curr.Time == p.Time {
```

Comment:
I would concur with @nathanielc's logic here, if this is meant to be a 'wrapper' function to something like derivative anyway, let the derivative handle the dupes accordingly and simply sum up the reseults it presents. That should keep things easy.